src/test/kotlin/no/nav/familie/klage/behandling/enhet/BehandlendeEnhetServiceTest.kt

```diff
@@ -0,0 +1,157 @@
+        @ParameterizedTest
+        @EnumSource(BarnetrygdEnhet::class, names = ["MIDLERTIDIG_ENHET"], mode = EnumSource.Mode.EXCLUDE)
+        fun `skal oppdatere behandlende enhet på behandling og oppgave samt opprette historikkinnslag`(
+            nyBehandlendeEnhet: BarnetrygdEnhet,
+        ) {
+            // Arrange
+            val fagsak = fagsak(stønadstype = Stønadstype.BARNETRYGD)
+            val behandling = behandling(fagsak = fagsak, behandlendeEnhet = BarnetrygdEnhet.DRAMMEN.enhetsnummer)
+            val begrunnelse = "Behandlende enhet var registrert på feil enhet"
+            val oppdaterBehandlendeEnhetRequest =
+                OppdaterBehandlendeEnhetRequest(
+                    enhetsnummer = nyBehandlendeEnhet.enhetsnummer,
+                    begrunnelse = begrunnelse,
+                )
+            val behandlendeEnhetSlot = slot<Enhet>()
+            val historikkHendelseSlot = slot<HistorikkHendelse>()
+            val beskrivelseSlot = slot<String>()
+            val behandlendeEnhetOppgaveSlot = slot<Enhet>()
+
+            every { behandlingService.hentBehandling(behandling.id) } returns behandling
+            every { fagsakService.hentFagsak(fagsak.id) } returns fagsak
+            every { behandlingService.oppdaterBehandlendeEnhet(behandling.id, capture(behandlendeEnhetSlot), Fagsystem.BA) } just Runs
+            every {
+                behandlingshistorikkService.opprettBehandlingshistorikk(
+                    behandlingId = behandling.id,
+                    steg = behandling.steg,
+                    historikkHendelse = capture(historikkHendelseSlot),
+                    beskrivelse = capture(beskrivelseSlot),
+                )
+            } returns mockk()
+
+            every { oppgaveService.oppdaterEnhetPåBehandleSakOppgave(behandling.id, capture(behandlendeEnhetOppgaveSlot)) } just Runs
+
+            // Act
+            behandlendeEnhetService.oppdaterBehandlendeEnhet(behandling.id, oppdaterBehandlendeEnhetRequest)
+
+            // Assert
+            verify(exactly = 1) { behandlingService.oppdaterBehandlendeEnhet(any(), any(), any()) }
+            verify(exactly = 1) { behandlingshistorikkService.opprettBehandlingshistorikk(any(), any(), any(), any()) }
+            verify(exactly = 1) { oppgaveService.oppdaterEnhetPåBehandleSakOppgave(any(), any()) }
+
+            assertThat(behandlendeEnhetSlot.captured).isEqualTo(nyBehandlendeEnhet)
+            assertThat(historikkHendelseSlot.captured).isEqualTo(HistorikkHendelse.BEHANDLENDE_ENHET_ENDRET)
+            assertThat(beskrivelseSlot.captured).contains(begrunnelse)
+            assertThat(behandlendeEnhetOppgaveSlot.captured).isEqualTo(nyBehandlendeEnhet)
+        }
+
+        @ParameterizedTest
+        @EnumSource(KontantstøtteEnhet::class, names = ["MIDLERTIDIG_ENHET"], mode = EnumSource.Mode.EXCLUDE)
+        fun `skal oppdatere behandlende enhet på behandling og oppgave samt opprette historikkinnslag`(
```

Comment:
Kanskje greit å skille på disse to?

```suggestion
        fun `skal oppdatere behandlende enhet på behandling og oppgave samt opprette historikkinnslag for KS`(
```